examples/pde.hpp

```diff
@@ -342,6 +342,141 @@ double HalfCoeffecient::Eval(mfem::ElementTransformation& T,
     return transip(1) < (spe10_scale_ * 44 * 5) ? -value_ : value_;
 }
 
+/**
+ @brief compute transmissibility based on two-point flux approximation
+
+ The implementation is based on (Sec. 4.4.1 of) the preliminary version of
+
+ K.-A. Lie. An introduction to reservoir simulation using MATLAB/GNU Octave
+ **/
+class LocalTPFA
+{
+    mfem::Coefficient* Q_;
+    mfem::VectorCoefficient* VQ_;
+    const mfem::ParMesh& mesh_;
+    const mfem::SparseMatrix& vert_edge_;
+
+    mfem::DenseMatrix EvalKappa(int i);
+    mfem::Vector ComputeShapeCenter(const mfem::Element& el);
+    mfem::Vector ComputeLocalWeight(int i);
+public:
+    LocalTPFA(const mfem::ParMesh& mesh, const mfem::SparseMatrix& vert_edge)
```

Comment:
I think this (and the other constructors) can build `vert_edge` from the `mesh`, as is currently done in `DarcyProblem::BuildReservoirGraph`.

It doesn't matter whether this is done here or in `DarcyProblem` (in fact I'm not too sure what the difference between them should be), but it would make this interface simpler.

Reply:
Yes, I also thought of doing that (was debating with myself). With your opinion now I know which option to choose.